docs/porting/custom-target-howto.md

```diff
@@ -0,0 +1,117 @@
+
+Let's get back to your application. We need to create your new target and tell both CMake and Mbed about it. We'll start with telling Mbed.
+
+### Telling Mbed about your custom board
+
+To tell Mbed about your target, create a new file called `custom_targets.json` in your application's directory.
+
+```
+mbed-os-example-blinky
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json     <--- The new file!
+├── LICENSE
+├── main.cpp
+├── mbed-os.lib
+├── README.md
+└── resources
+    └── official_armmbed_example_badge.png
+```
+
+Inside this new file, you can use Mbed's configuration system to specify extra information about your board. An example is below. [Our target documentation](https://os.mbed.com/docs/mbed-os/latest/program-setup/adding-and-configuring-targets.html) provides further information on what configuration options are available for targets.
+
+```
+{
+    "IMAGINARY": {
+        "inherits": [
+            "NRF52840"
+        ]
+    }
+}
+```
+
+#### Making the CMake target
+
+Next, we need to tell CMake about your custom target.
+
+The most common thing to add for custom targets is a custom pin configuration. As such, we'll assume you'll be wanting to customize your custom board's pin configuration too. To do this, you'll need to make a few files: at least a `PinNames.h` and possibly also a `PeripheralPins.h` or a `PeripheralNames.h`. Please refer to [the Mbed OS porting guide](https://os.mbed.com/docs/mbed-os/latest/porting/porting-custom-boards.html) on how to configure the pins for your board and which files exactly you'd need to create.
+
+```
+mbed-os-example-blinky
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json
+├── IMAGINARY              <--- The new folder with new files in it!
+    ├── CMakeLists.txt
+    ├── PeripheralNames.h
+    ├── PeripheralPins.h
+    └── PinNames.h
+├── LICENSE
+├── main.cpp
+├── mbed-os.lib
+├── README.md
+└── resources
+    └── official_armmbed_example_badge.png
+```
+
+Finally, we tell CMake about your new custom board. We do this by making a new CMakeLists.txt, separate from your application's CMakeLists.txt, which describes your target and how to build for it.
+
+```
+mbed-os-example-blinky
+├── CMakeLists.txt
+├── CONTRIBUTING.md
+├── custom_targets.json
+├── IMAGINARY
+    ├── CMakeLists.txt     <--- The new file!
+    ├── PeripheralNames.h
+    ├── PeripheralPins.h
+    └── PinNames.h
+├── LICENSE
+├── main.cpp
+├── mbed-os.lib
+├── README.md
+└── resources
+    └── official_armmbed_example_badge.png
+```
+
+The CMakeLists.txt file should have at a minimum a new library target, as well as a list of which files or include directories your target needs. The [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) directive should specify which MCU CMake target your custom board depends on, so you'll enter here the MCU Cmake target name you noted previously.
```

Comment:
```suggestion
The ./IMAGINARY/CMakeLists.txt file should have at a minimum a new library target, as well as a list of which files or include directories your target needs. The [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) directive should specify which MCU CMake target your custom board depends on, so you'll enter here the MCU Cmake target name you noted previously.
```

Reply:
Would adding the folder name helpful?

Reply:
Good idea

Reply:
I've removed this file, so this probably isn't relevant anymore.